contracts/facets/ThirdPartyFacet.sol

```diff
@@ -0,0 +1,254 @@
+                LibTypes.ArchaeologistStorage memory defaulter = s
+                    .sarcophagusArchaeologists[identifier][archAddresses[i]];
+
+                totalBounty += defaulter.bounty;
+                totalDiggingFee += defaulter.diggingFee;
+
+                uint256 cursedBond = LibBonds.calculateCursedBond(
+                    defaulter.diggingFee,
+                    defaulter.bounty
+                );
+
+                totalCursedBond += cursedBond;
+
+                // decrease the defaulter's cursed bond
+                LibBonds.decreaseCursedBond(archAddresses[i], cursedBond);
+            }
+        }
+
+        (
+            uint256 cleanerBondReward,
+            uint256 embalmerBondReward
+        ) = _distributeLoot(
+                paymentAddress,
+                sarco,
+                totalCursedBond,
+                totalDiggingFee,
+                totalBounty
+            );
+
+        emit LibEvents.CleanUpSarcophagus(
+            identifier,
+            msg.sender,
+            cleanerBondReward,
+            embalmerBondReward
+        );
+    }
+
+    function accuse(
+        bytes32 sarcoId,
+        bytes32[] memory unencryptedShards,
+        address paymentAddress
+    ) external {
+        LibTypes.Sarcophagus storage sarco = s.sarcophaguses[sarcoId];
+        if (unencryptedShards.length < sarco.minShards) {
+            revert LibErrors.NotEnoughProof();
+        }
+
+        address[] memory bondedArchsAddresses = sarco.archaeologists;
+
+        LibTypes.ArchaeologistStorage[]
+            memory accusedArchs = new LibTypes.ArchaeologistStorage[](
+                unencryptedShards.length
+            );
+
+        // For each provided shard, search through each archaeologist's hashed shard
+        // to see if the provided hard's hash matches one on storage. If so, flag that
+        // archaeologist as accusable
+        uint256 pos = 0;
+        for (uint256 i = 0; i < unencryptedShards.length; i++) {
+            bytes32 shardHash = _hashHelper(unencryptedShards[i]);
+            bool matchingHash = false;
+
+            // Ew
+            for (uint256 j = 0; j < bondedArchsAddresses.length; j++) {
+                LibTypes.ArchaeologistStorage storage bondedArch = s
+                    .sarcophagusArchaeologists[sarcoId][
+                        bondedArchsAddresses[j]
+                    ];
+
+                if (bondedArch.hashedShard == shardHash) {
+                    accusedArchs[pos++] = bondedArch;
+                    matchingHash = true;
+                    break;
+                }
+            }
+
+            if (!matchingHash) {
+                revert LibErrors.NotEnoughProof();
+            }
+        }
+
+        // Reaching this point means accusal is good to go
+
+        uint256 diggingFeesToBeDistributed = 0;
+        uint256 bountyToBeDistributed = 0;
+        uint256 totalCursedBond = 0;
+
+        for (uint256 i = 0; i < accusedArchs.length; i++) {
```

Comment:
Could the contents of this for loop be combined with the previous loop instead of having two loops?